diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/SortRule.kt

```diff
@@ -71,11 +78,47 @@ class SortRule(configRules: List<RulesConfig>) : DiktatRule("sort-rule", configR
         sortList: List<ASTNode>,
         nonSortList: List<ASTNode>,
         node: ASTNode) {
+        val spaceBefore = nonSortList.map { astNode ->
+            val nodeBefore = astNode.prevSibling { it.elementType == WHITE_SPACE || it.elementType == ENUM_ENTRY }
+            if (nodeBefore?.elementType == WHITE_SPACE) {
+                nodeBefore
+            } else {
+                null
+            }
+        }
         val nodeBefore: ASTNode? = nonSortList.last().treeNext
         node.removeRange(nonSortList.first(), nonSortList.last().treeNext)
         sortList.forEachIndexed { nodeIndex, astNode ->
-            if (nodeIndex != 0) {
-                node.addChild(PsiWhiteSpaceImpl("\n"), nodeBefore)
+            spaceBefore[nodeIndex]?.let { node.addChild(it, nodeBefore) }
+            node.addChild(astNode, nodeBefore)
+        }
+    }
+
+    private fun swapSortEnumEntry(
+        sortList: List<ASTNode>,
+        nonSortList: List<ASTNode>,
+        node: ASTNode) {
+        val spaceBefore = if (node.findAllNodesWithSpecificType(EOL_COMMENT).isNotEmpty()) {
+            val lastEntry = node.findAllNodesWithSpecificType(ENUM_ENTRY).last()
+            if (lastEntry.hasChildOfType(EOL_COMMENT) && !lastEntry.hasChildOfType(COMMA)) {
+                lastEntry.addChild(LeafPsiElement(COMMA, ","), lastEntry.findChildByType(EOL_COMMENT))
+            }
+            listOf(null) + List(nonSortList.size - 1) { listOf(PsiWhiteSpaceImpl("\n")) }
```

Comment:
Maybe, use `buildList` instead, or at least `repeat` instead of `List`? Also, you can check `if (index == 0)` inside second argument of `List`, it'll be better than lists concat